controllers/spacerequest/spacerequest_controller.go

```diff
@@ -43,12 +63,440 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[strin
+// returns true if the conditions list changed
+// returns false if the conditions list is unchanged
+func reflectSubSpaceReadyCondition(subSpace *toolchainv1alpha1.Space, spaceRequest *toolchainv1alpha1.SpaceRequest) bool {
+	var addedOrUpdated bool
+	if condition.IsTrue(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady) {
+		// subSpace was provisioned so let's set Ready type condition to true on the space request
+		spaceRequest.Status.Conditions, addedOrUpdated = condition.AddOrUpdateStatusConditions(spaceRequest.Status.Conditions, toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.ConditionReady,
+			Status: v1.ConditionTrue,
+			Reason: toolchainv1alpha1.SpaceProvisionedReason,
+		})
+	} else if condition.IsFalse(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady) {
+		// subSpace is not ready let's copy the condition together with reason/message fields
+		conditionReadyFromSubSpace, _ := condition.FindConditionByType(subSpace.Status.Conditions, toolchainv1alpha1.ConditionReady)
+		spaceRequest.Status.Conditions, addedOrUpdated = condition.AddOrUpdateStatusConditions(spaceRequest.Status.Conditions, conditionReadyFromSubSpace)
+	}
+	return addedOrUpdated
+}
+
+// setFinalizers sets the finalizers for Space
+func (r *Reconciler) addFinalizer(logger logr.Logger, memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest) error {
+	// Add the finalizer if it is not present
+	if !util.HasFinalizer(spaceRequest, toolchainv1alpha1.FinalizerName) {
+		logger.Info("adding finalizer on SpaceRequest")
+		util.AddFinalizer(spaceRequest, toolchainv1alpha1.FinalizerName)
+		if err := memberCluster.Client.Update(context.TODO(), spaceRequest); err != nil {
+			return errs.Wrap(err, "error while adding finalizer")
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) ensureSpace(logger logr.Logger, memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest) (*toolchainv1alpha1.Space, bool, error) {
+	logger.Info("ensuring Space")
+
+	// get subSpace resource
+	spaceList, err := r.listSubSpaces(spaceRequest)
+	if err != nil {
+		return nil, false, err
+	}
+
+	// validate tierName
+	if err := r.validateNSTemplateTier(spaceRequest.Spec.TierName); err != nil {
+		return nil, false, err
+	}
+
+	// find parent space from namespace labels
+	parentSpace, err := r.getParentSpace(memberCluster, spaceRequest)
+	if err != nil {
+		return nil, false, err
+	}
+	// parentSpace is being deleted
+	if util.IsBeingDeleted(parentSpace) {
+		return nil, false, errs.New("parentSpace is being deleted")
+	}
+
+	var subSpace *toolchainv1alpha1.Space
+	switch {
+	case len(spaceList.Items) == 0:
+		// no spaces found, let's create it
+		subSpace, err = r.createNewSubSpace(logger, spaceRequest, parentSpace)
+		if err != nil {
+			// failed to create subSpace
+			return subSpace, false, r.setStatusFailedToCreateSubSpace(logger, memberCluster, spaceRequest, err)
+		}
+		return subSpace, true, nil // a subSpace was created
```

Comment:
correct me if I missed anything - when we created the Space CR, then we should update the condition in SR to "provisioning", but I don't see anything like that anywhere.

Reply:
added please see : [60159bc](https://github.com/codeready-toolchain/host-operator/pull/770/commits/60159bc49fc4872acb0b880853cb3775e5454300)

It was not there initially because it is also reflected from the Space CR